Added support for MultiTarget with different token

Coder-ZJQ opened the pull request:
Changed the parameter type `AuthorizationType` to `TargetType` in the token closure to add support for `MultiTarget` with different tokens:
``` swift
let fooToken = "eyeAm.AJsoN.weBTOKen.foo"
let barToken = "eyeAm.AJsoN.weBTOKen.bar"
let authPlugin = AccessTokenPlugin { target in 
    if target is FooService {
        return fooToken    
    } else if target is BarService {
        return barToken
    }
    return ""
}
let provider = MoyaProvider<MultiTarget>(plugins: [authPlugin])
```

<!--
Thank you for contributing to Moya! 🙌


Choosing a base branch:

  master: bug fixes, non breaking API changes, documentation fixes
  development: breaking changes, features for the next version


If your pull request fixes an issue, please reference the issue.
For example, when your pull request fixes issue 10, add the following line:

Fixes #10

This will make sure that when the pull request is merged, the issue will automatically be closed.

-->

sunshinejr commented:
Thanks for taking your time and issueing a PR @Coder-ZJQ (also sorry for a delayed review) - this looks cool! I can see a use-case where you would have different tokens for different targets, e.g. using facebook/google sign-ins.

This is, however, another breaking change to the auth plugin so I would like a double-check with @Moya/contributors if there is anything we should consider before merging it.

Coder-ZJQ commented:
It's my pleasure : ) And I have added the [changelog entry](https://github.com/Moya/Moya/pull/2046/files#diff-9b8fab691c00b9e5380b19ce882f3271), the docs([docs_CN/Authentication.md](https://github.com/Moya/Moya/pull/2046/files#diff-60b4b40900ca70b12445864bab156f9a), [docs/Authentication.md](https://github.com/Moya/Moya/pull/2046/files#diff-c3ae69f3d76aad8177df8dd3a54c9476)) I have updated in the first commit. @sunshinejr

sunshinejr commented:
@Coder-ZJQ thank you so much for working on this, I've added few docs changes but this is great and will be in the next beta 🙏 (sorry for the delay on that as well...)
